airbyte-cdk/python/airbyte_cdk/sources/declarative/manifest_declarative_source.py

```diff
@@ -142,6 +144,33 @@ def _validate_source(self):
         except ValidationError as e:
             raise ValidationError("Validation against json schema defined in declarative_component_schema.yaml schema failed") from e
 
+        cdk_version = metadata.version("airbyte_cdk")
+        cdk_major, cdk_minor, cdk_patch = (int(part) for part in cdk_version.split("."))
```

Comment:
why not using `_get_version_parts` here?

Reply:
we could, the main value of `_get_version_parts()` was it does some error checking since manifests could have any string, whereas for the package string is "guaranteed". That being said, its easy to reuse the two and still ensure a good error message so i'll fix this